File: assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/FSNormalizeTest1.cs
using Xunit;

namespace DefaultNamespace
{
    public class FileSystemNormalizeTests
    {
        [Fact]
        public void FSNormalizeTest1()
        {
            {
                ScopeContext.PushProperty("foo", "abc.log");
                SimpleLayout l = "${filesystem-normalize:${scopeproperty:foo}}";
                Assert.Equal("abc.log", l.Render(LogEventInfo.CreateNullEvent()));
                ScopeContext.PushProperty("foo", "");
                Assert.Equal("", l.Render(LogEventInfo.CreateNullEvent()));
                ScopeContext.PushProperty("foo", "a/b/c");
                Assert.Equal("a_b_c", l.Render(LogEventInfo.CreateNullEvent()));
                // all characters outside of alpha/digits/space/_/./- are replaced with _
                ScopeContext.PushProperty("foo", ":\\/$@#$%^");
                Assert.Equal("_________", l.Render(LogEventInfo.CreateNullEvent()));
            }
        }
    }
}

File: smelly/assertion-roulette/java-junit/FSNormalizeTest1.java
import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class FileSystemNormalizeTests {

    @Test
    public void FSNormalizeTest1() {
        {
            ScopeContext.pushProperty("foo", "abc.log");
            SimpleLayout l = "${filesystem-normalize:${scopeproperty:foo}}";
            assertEquals("abc.log", l.render(LogEventInfo.createNullEvent()));
            ScopeContext.pushProperty("foo", "");
            assertEquals("", l.render(LogEventInfo.createNullEvent()));
            ScopeContext.pushProperty("foo", "a/b/c");
            assertEquals("a_b_c", l.render(LogEventInfo.createNullEvent()));
            ScopeContext.pushProperty("foo", ":\\/$@#$%^");
            assertEquals("_________", l.render(LogEventInfo.createNullEvent()));
        }
    }
}
